XCUITests/TranslationSnackBarTest.swift

```diff
@@ -5,15 +5,41 @@
 import XCTest
 
 class TranslationSnackBarTest: BaseTestCase {
-    func testSnackBar() {
+    
+    // This test checks to the correct functionalty of the Translation prompt and Translation is done corrrectly using Google
+    func testSnackBarDisplayed() {
         userState.localeIsExpectedDifferent = true
         navigator.openURL(path(forTestPage: "manifesto-zh-CN.html"))
         waitUntilPageLoad()
-        navigator.goto(TranslatePageMenu)
         waitForExistence(app.buttons["TranslationPrompt.doTranslate"])
+        app.buttons["TranslationPrompt.dontTranslate"].tap()
+        XCTAssertFalse(app.buttons["TranslationPrompt.dontTranslate"].exists)
+        navigator.performAction(Action.ReloadURL)
+        waitForExistence(app.buttons["TranslationPrompt.doTranslate"])
+        navigator.performAction(Action.SelectTranslateThisPage)
+        waitForValueContains(app.textFields["url"], value: "translate.google")
+        XCUIApplication().buttons["Reload"].tap()
     }
-
-    func testSetting() {
+    
+    // This test checks to see if Translation is enabled by default from the Settings menu and can be correctly disabled
+    func testTranslationDisabled() {
         navigator.goto(TranslationSettings)
+        let translationSwitch = app.switches["TranslateSwitchValue"]
+        XCTAssertTrue(translationSwitch.isEnabled)
+        translationSwitch.tap()
+        navigator.openURL(path(forTestPage: "manifesto-zh-CN.html"))
+        waitUntilPageLoad()
+        XCTAssertFalse(app.buttons["TranslationPrompt.dontTranslate"].exists)
+    }
+    
+    // This test checks to see if Translation is correctly done when using Bing
+    func testTranslateBing() {
+        navigator.goto(TranslationSettings)
+        app.tables["Translate.Setting.Options"].staticTexts["Bing"].tap()
+        navigator.openURL(path(forTestPage: "manifesto-zh-CN.html"))
+        waitUntilPageLoad()
+        waitForExistence(app.buttons["TranslationPrompt.doTranslate"])
+        app.buttons["TranslationPrompt.doTranslate"].tap()
```

Comment:
I think you can use the action defined for this: `navigator.performAction(Action.SelectTranslateThisPage)`